x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -505,11 +506,31 @@ export class Authenticator {
    * @param request Request instance.
    */
   async reauthenticate(request: KibanaRequest) {
-    assertRequest(request);
+    // "Fake" requests cannot be re-authenticated as there are no sessions associated with them.
+    if (!(request instanceof CoreKibanaRequest)) {
+      this.logger.debug(
+        `Re-authentication is only supported for instances of \`CoreKibanaRequest\`, but got [\`${typeof request}\`] with keys [${
+          request ? Object.keys(request) : []
+        }].`
+      );
+      return AuthenticationResult.notHandled();
+    }
```

Comment:
__question:__ @elastic/kibana-core would it make sense to perform this check inside the core instead? So that you don't even call handler registered with `elasticsearch.setUnauthorizedErrorHandler` for fake requests since we cannot "re-authenticate" them anyway/by-design.

Reply:
AFAIK "Fake" requests at this level are instances of `CoreKibanaRequest` anyway... I don't think we have any other concrete class implementing `KibanaRequest`, do we? So I'm not sure what this check is even asserting against tbh (and I'm not sure I want to know, but you will probably tell me anyway)

Reply:
> AFAIK "Fake" requests at this level are also instances of CoreKibanaRequest anyway.

Apparently not all of them :slightly_smiling_face: I'll share the findings from Overview Cluster in private.

Reply:
As we discussed over Slack, I removed `assertRequest` completely in https://github.com/elastic/kibana/pull/176503/commits/99e98d19bd4c1443714fe2a9bef10dc9219f741f since the proper request type should be guaranteed by the TypeScript and Core these days :slightly_smiling_face: